administration/src/bp-modules/components/CustomDatePicker.tsx

```diff
@@ -1,55 +1,81 @@
 import { DesktopDatePicker } from '@mui/x-date-pickers'
 import { deDE } from '@mui/x-date-pickers/locales'
-import React, { ReactElement } from 'react'
+import React, { CSSProperties, ReactElement } from 'react'
 
 import useWindowDimensions from '../../hooks/useWindowDimensions'
 
 type CustomDatePickerProps = {
-  date: Date | null
-  onBlur: () => void
-  onChange: (date: Date | null) => void
-  onClear: () => void
+  value: Date | null
+  disabled?: boolean
+  label?: string
+  onBlur?: () => void
+  onChange?: (date: Date | null) => void
+  onClear?: () => void
   isValid: boolean
   minDate?: Date
   maxDate?: Date
-  disableFuture: boolean
+  disableFuture?: boolean
+  textFieldHelperText?: string
+  textFieldStyle?: CSSProperties
 }
 
 const CustomDatePicker = ({
-  date,
+  value,
+  disabled,
+  label,
   onBlur,
   onChange,
   onClear,
   isValid,
   minDate,
   maxDate,
   disableFuture,
+  textFieldHelperText,
+  textFieldStyle,
 }: CustomDatePickerProps): ReactElement => {
   const { viewportSmall } = useWindowDimensions()
-  const formStyle = viewportSmall ? { fontSize: 16, padding: '9px 10px' } : { fontSize: 14, padding: '6px 10px' }
-  const textFieldBoxShadow =
-    '0 0 0 0 rgba(205, 66, 70, 0), 0 0 0 0 rgba(205, 66, 70, 0), inset 0 0 0 1px #cd4246, inset 0 0 0 1px rgba(17, 20, 24, 0.2), inset 0 1px 1px rgba(17, 20, 24, 0.3)'
+
   return (
     <DesktopDatePicker
+      disabled={disabled}
+      label={label}
       views={['year', 'month', 'day']}
-      value={date}
+      openTo='year'
+      value={value}
       format='dd.MM.yyyy'
+      sx={{ '& input[value=""]:not(:focus)': { color: 'transparent' } }}
       slotProps={{
-        clearIcon: { fontSize: viewportSmall ? 'medium' : 'small' },
         openPickerIcon: { fontSize: 'small' },
-        field: { clearable: true, onClear },
+        field: {
+          clearable: true,
+          onClear,
+        },
         textField: {
+          helperText: textFieldHelperText,
           placeholder: 'TT.MM.JJJJ',
           error: !isValid,
           spellCheck: false,
           onBlur,
+          style: textFieldStyle,
+          size: 'small',
           sx: {
             width: '100%',
-            input: formStyle,
-            boxShadow: !isValid ? textFieldBoxShadow : undefined,
+            boxShadow: !isValid
+              ? `
+                0 0 0 0 rgba(205, 66, 70, 0),
+                inset 0 0 0 1px #cd4246,
+                inset 0 0 0 1px rgba(17, 20, 24, 0.2),
+                inset 0 1px 1px rgba(17, 20, 24, 0.3),
+              `
+              : undefined,
+            '.MuiPickersInputBase-root': {
+              fontSize: viewportSmall ? '16px' : '14px',
+              borderRadius: '2px',
```

Comment:
add here `backgroundColor: 'white'` that is required for the `filled` variant to align the background styling

Reply:
I added some code to accommodate all kind of styles to match the requested style (see the last commit in this PR). However, MUI is clearly not designed for this. I think at some point we need to setup a way to have one theme for all MUI components per customer.